include/wil/result_macros.h

```diff
@@ -569,24 +569,32 @@ WI_ODR_PRAGMA("WIL_FreeMemory", "0")
 #define __RFF_FN(MethodName)                                  __RFF_NS:: MethodName
 #endif
 // end-of-repeated fail-fast handling macros
+
+#if defined(_MSVC_LANG)
+#define __WI_SUPPRESS_4127_S __pragma(warning(push)) __pragma(warning(disable:4127))

```

Comment:
Just curious, why not simplify with __pragma(warning(suppress:4127)), which wouldn't need an end tag?

Reply:
The [documentation for warning-suppress](https://docs.microsoft.com/en-us/cpp/preprocessor/warning?view=vs-2019) indicates that the `#pragma` applies to the _following_ line. The `__pragma` form appears to suppress it on both this line and the following line:

https://godbolt.org/z/12GZkp

```c++
const int kitten = 6;
const int poptart = 6;
const int muffin = 7;
void toast();
void foo() {
    __pragma(warning(suppress:4127)) if (kitten == poptart) { toast(); }
    if (muffin == poptart) { toast(); }
    if (muffin == kitten) { toast(); }
}
```

Only the `muffin == kitten` line shows an error, when `muffin == poptart` should also show an error.